terminus/builders/osm_city_builder.py

```diff
@@ -210,3 +239,59 @@ def _translate_coords(self, lat, lon):
         coords = LatLon(lat, lon)
         (delta_lat, delta_lon) = self.map_origin.delta_in_meters(coords)
         return Point(delta_lon, delta_lat, 0)
+
+    def _check_intersection_with_bounding_box(self, prev_coord, coord):
+        '''
+        Checks if a road segment (provided 2 coordinates for the nodes) intersects the bounding box
+        '''
+        intersections = []
+        # Check if segment intersects with bounding box
+        intersections.append(self._find_intersection(prev_coord,
+                                                     coord,
+                                                     self.bounds['min_x_max_y'],
+                                                     self.bounds['max_xy']))
+        intersections.append(self._find_intersection(prev_coord,
+                                                     coord,
+                                                     self.bounds['min_x_max_y'],
+                                                     self.bounds['min_xy']))
+        intersections.append(self._find_intersection(prev_coord,
+                                                     coord,
+                                                     self.bounds['max_x_min_y'],
+                                                     self.bounds['max_xy']))
+        intersections.append(self._find_intersection(prev_coord,
+                                                     coord,
+                                                     self.bounds['max_x_min_y'],
+                                                     self.bounds['min_xy']))
+        return [intersection for intersection in intersections if intersection is not None]
+
+    def _find_intersection(self, p0, p1, p2, p3):
```

Comment:
Once https://github.com/ekumenlabs/terminus/pull/154 is merged, can you move this implementation to the `LineSegment` class and add a couple of tests? This way we can start reusing this things.